src/workspacer.Native/Native/WindowsManager.cs

```diff
@@ -30,10 +30,31 @@ public class WindowsManager : IWindowsManager
 
         private Dictionary<WindowsWindow, bool> _floating;
 
+        /// <summary>
+        /// Notifies when a new window handle was created by the manager
+        /// </summary>
         public event WindowCreateDelegate WindowCreated;
+        /// <summary>
+        /// Notifies when a handled window was removed by the manager
+        /// </summary>
         public event WindowDelegate WindowDestroyed;
+        /// <summary>
+        /// Notifies when a handled window was updated by the manager
+        /// </summary>
         public event WindowUpdateDelegate WindowUpdated;
+
+        /// <summary>
+        /// Notifies when a window focuses itself
+        /// </summary>
         public event WindowFocusDelegate WindowFocused;
+        /// <summary>
+        /// Notifies when a window updated itself
+        /// </summary>
+        public event WindowDelegate OnWindowUpdated;
```

Comment:
In what ways is this event different enough from the existing `WindowUpdated` event that it needs to be special cased?

Just curious! :smile:

Reply:
Right. Reading the code, it seems we have this event specifically to be able to further trigger manager-based `WindowUpdated` events without causing recursion and event-looping?

If so, that's fine by itself. I'm just wondering if there would be a way to structure the code to allow this without having two seemingly similar events like this. Any opinions?

Reply:
As you indeed correctly, using the existing event in its current state causes a feedback loop of updates (found that out the hard way). As the logic path for this event is very specific to processing updates on a window within a workspace I think merging the two would cause more confusion, though the suggestion would be adding another bool param to the event like "doWindowHandling" with default true which would skip the handling code when false.

Reply:
If we either make it clear that one event is for *internal* changes or that the other event is for *externally* triggered changes.... That would be a good enough distinction for me.

Since one event already was there from the get go, I think it makes sense to tag this new event as being the "external" one.

It's a bit verbose and rich... But how about `WorkspacerExternalWindowsUpdate`? It's only registered one place after all?

Reply:
I liked the suggested fix so i applied it!